public/controllers/register-agent/containers/steps/steps.tsx

```diff
@@ -1,143 +1,104 @@
 import {
-  REGISTER_AGENT_DATA_STEP_THREE,
-  REGISTER_AGENT_DATA_STEP_TWO,
-} from '../../utils/register-agent-data';
-import { webDocumentationLink } from '../../../../../common/services/web_documentation';
-import { PLUGIN_VERSION_SHORT } from '../../../../../common/constants';
-
-const popoverServerAddress = (
-  <span>
-    Learn about{' '}
-    <EuiLink
-      href={webDocumentationLink(
-        'user-manual/reference/ossec-conf/client.html#groups',
-        PLUGIN_VERSION_SHORT,
-      )}
-      target='_blank'
-      rel='noopener noreferrer'
-    >
-      Server address.
-    </EuiLink>
-  </span>
-);
-
-const popoverAgentName = (
-  <span>
-    Learn about{' '}
-    <EuiLink
-      href={webDocumentationLink(
-        'user-manual/reference/ossec-conf/client.html#enrollment-agent-name',
-        PLUGIN_VERSION_SHORT,
-      )}
-      target='_blank'
-      rel='noopener noreferrer'
-    >
-      Assigning an agent name.
-    </EuiLink>
-  </span>
-);
+  IParseRegisterFormValues,
+  getAgentCommandsStepStatus,
+  getOSSelectorStepStatus,
+  getOptionalParameterStepStatus,
+  getRegisterAgentFormValues,
+  getServerAddressStepStatus,
+  parseRegisterAgentFormValues,
+  showCommandsSections,
+} from '../../services/register-agent-services';
+import { useRegisterAgentCommands } from '../../hooks/use-register-agent-commands';
+import {
+  osCommandsDefinitions,
+  optionalParamsDefinitions,
+  tOperatingSystem,
+  tOptionalParameters,
+} from '../../core/config/os-commands-definitions';
+import { UseFormReturn } from '../../../../components/common/form/types';
+import CommandOutput from '../../components/command-output/command-output';
+import ServerAddressTitle from '../../components/server-address/server-address-title';
+import ServerAddressInput from '../../components/server-address/server-address-input';
+import OptionalsInputs from '../../components/optionals-inputs/optionals-inputs';
 
-const warningForAgentName =
-  'The agent name must be unique. It can’t be changed once the agent has been enrolled.';
+interface IStepsProps {
+  needsPassword: boolean;
+  hideTextPassword: boolean;
+  form: UseFormReturn;
+  osCard: React.ReactElement;
+  connection: {
+    isUDP: boolean;
+    isSecure: boolean;
+  };
+  wazuhPassword: string;
+}
 
 export const Steps = ({
   needsPassword,
   hideTextPassword,
-  agentGroup,
   form,
   osCard,
-}) => {
-  const [isPopoverServerAddress, setIsPopoverServerAddress] = useState(false);
-  const [isPopoverAgentName, setIsPopoverAgentName] = useState(false);
+  connection,
+  wazuhPassword,
+}: IStepsProps) => {
+  const [registerAgentFormValues, setRegisterAgentFormValues] =
+    useState<IParseRegisterFormValues>({
+      operatingSystem: {
+        name: '',
+        architecture: '',
+      },
+      optionalParams: {
+        agentGroups: '',
+        agentName: '',
+        serverAddress: '',
+        wazuhPassword,
+      },
+    });
```

Comment:
**thought**: The form state is controlled, why do we need to handle of this way? Can not we use the form state instead?

Reply:
This is because the Command Generator i need to pass only the values to the Command Generator. And must be passed separated by OS and optionals params.